### ISSUE TITLE
Indexing Error - boost::bad_alloc

### ISSUE THREAD

**Issue by cloudburst** [CoatiSoftware/SourcetrailBugTracker#396]
This error is on Linux 64 bit builds of `Sourcetrail 0.13.0`  I am attempting to build a large C++ project with a compilation database of clang++ commands.

```
10:53:38 | StatusBarController.cpp:74 setStatus() | INFO: STATUS Preparing Project: Processing Files
10:54:05 | StatusBarController.cpp:74 setStatus() | INFO: STATUS Reindexing Project
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "edge_multipart_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "node_serialized_name_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "local_symbol_name_index"
10:54:10 | SqliteDatabaseIndex.cpp:34 removeFromDatabase() | INFO: Removing database index "source_location_file_node_id_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "source_location_all_data_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "comment_location_all_data_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "error_all_data_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "file_path_index"
10:54:10 | SqliteDatabaseIndex.cpp:34 removeFromDatabase() | INFO: Removing database index "occurrence_element_id_index"
10:54:10 | SqliteDatabaseIndex.cpp:34 removeFromDatabase() | INFO: Removing database index "occurrence_source_location_id_index"
10:54:10 | SqliteDatabaseIndex.cpp:19 createOnDatabase() | INFO: Creating database index "component_access_node_id_index"
10:54:10 | InterprocessIndexerCommandManager.cpp:39 setIndexerCommands() | INFO: grow memory - est: 439338828 size: 1048576 free: 1048352 alloc: 438290476
10:54:12 | TaskRunner.cpp:30 update() | ERROR: boost::interprocess::bad_alloc
10:54:12 | INFO: send MessageStatus An unknown exception was thrown during indexing. - error
10:54:12 | StatusBarController.cpp:74 setStatus() | INFO: STATUS An unknown exception was thrown during indexing.
```

Due to this error, the project fails to index any files and just exits with that error.  If I try with the older `Sourcetrail 0.12.25` it just silently fails to index any files with the same configuration.

This is on a build box with 64GB of RAM that can successfully build the project fine outside of Coati so the issue is not a problem with the host available memory as far as I know.

**Comment by egraether** [comment]
Thanks for reporting! From the log it looks as if allocating 439338828 Bytes of shared memory fails. We might run into a shared memory page size maximum here. We will have a look on how to fix this.

**Comment by cloudburst** [comment]
Here is how to reproduce this error on Ubuntu Linux:

**Install depot_tools**
```
git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
export PATH="$PATH:/path/to/depot_tools"
```
**Generate JSON**
```
mkdir ~/chromium && cd ~/chromium
fetch --nohooks chromium
cd src
./build/install-build-deps.sh
gclient runhooks
gn gen out/Default
ninja -C out/Default -t compdb cc cxx > compdb.json
```
Then load compdb.json into Coati and the error instantly comes up.
 * https://chromium.googlesource.com/chromium/src/+/master/docs/linux_build_instructions.md

**Comment by mlangkabel** [comment]
We have been able to reproduce the issue on the chromium project. Fixed with Sourcetrail version 0.13.8 :)
